pkg/tsdb/elasticsearch/response_parser.go

```diff
@@ -592,3 +597,50 @@ func getErrorFromElasticResponse(response *es.SearchResponse) plugins.DataQueryR
 
 	return result
 }
+
+func processTopMetricValues(stats *simplejson.Json, metric *MetricAgg, field string) null.Float {
+	for _, _stat := range stats.MustArray() {
+		stat := _stat.(map[string]interface{})
+		_metrics, hasMetrics := stat["metrics"]
+		if hasMetrics {
+			metrics := _metrics.(map[string]interface{})
+			_metricValue, hasMetricValue := metrics[field]
+			if hasMetricValue && _metricValue != nil {
+				return null.FloatFrom(_metricValue.(float64))
+			}
+		}
+	}
+	return null.NewFloat(0, false)
+}
+
+func processTopMetrics(metric *MetricAgg, esAgg *simplejson.Json, props map[string]string) plugins.DataTimeSeriesSlice {
+	var series plugins.DataTimeSeriesSlice
+	_metrics, hasMetrics := metric.Settings.MustMap()["metrics"].([]interface{})
+	if hasMetrics {
+		metrics := make([]string, len(_metrics))
+		for i, v := range _metrics {
+			metrics[i] = v.(string)
+		}
```

Comment:
The intermediate `metrics` array is not needed, along with this for loop. Can handle the `v.(string)` every time you get it without having to store it.